neqo-transport/src/connection.rs

```diff
@@ -2972,4 +2972,153 @@ mod tests {
         let server_out = server.process(client_fin.dgram(), now());
         assert!(server_out.as_dgram_ref().is_some());
     }
+
+    #[test]
+    fn pto_works_basic() {
+        let mut client = default_client();
+        let mut server = default_server();
+        connect(&mut client, &mut server);
+
+        let now = now();
+
+        let res = client.process(None, now);
+        assert_eq!(res, Output::Callback(Duration::from_secs(60)));
+
+        // Send data on two streams
+        assert_eq!(client.stream_create(StreamType::UniDi).unwrap(), 2);
+        assert_eq!(client.stream_send(2, b"hello").unwrap(), 5);
+        assert_eq!(client.stream_send(2, b" world").unwrap(), 6);
+
+        assert_eq!(client.stream_create(StreamType::UniDi).unwrap(), 6);
+        assert_eq!(client.stream_send(6, b"there!").unwrap(), 6);
+
+        // Send orig pkt
+        let _out = client.process(None, now + Duration::from_secs(10));
```

Comment:
When we do these tests, would it be better to use a pattern of:

```rust
let mut now = now();

client.process(None, now);
// ....
now += Duration::from_secs(10);
client.process(None, now);
// ...
now += Duration::from_secs(1);
```
?

There's a risk with the current approach that we go backwards in time.  Which I guess is fine if that is the intent, but this seems clearer.

Happy to address this in a separate patch though.

Reply:
Good idea.